src/Http/GenericRouter.php

```diff
@@ -48,6 +49,8 @@ public function dispatch(Request $request): Response
 
         $callable = $this->getCallable($matchedRoute);
 
+        $request = $this->prepareRequest($request, $matchedRoute);
```

Comment:
Ok so, this is the part that needed serious refactoring.

Here's what happened. I wanted to test request validation, which is handled by the ObjectMapper — it's how all validation is done in tempest.

However, our router is set up in a way that it expects an already created and valid request class. In real life apps, this would happen in `RequestInitializer`, which was called actually in two places:

- Once in the `HttpApplication`, which would always create a `GenericRequest`
- A second time when calling the controller action via the container within the `Router`

So far so good, if there were any validation errors, that container call to the controller action would throw that error (via the initializer, which used the mapper). 

However, we skipped all of that during testing, since we manually registered a singleton for our concrete request object in the `HttpRouterTester`.

So how to solve that? I could reimplement all the validation logic in the `HttpRouterTester` as well, but that's just stupid because now I'm not testing real validation anymore.

This made me realise that I was on the wrong path trying to use Initializers for request. What needs to be done is that the `Router` gets a generic request object (yes, this _could_ be a PSR-7 object), and the router should take care of transforming it into the right request object and validating it. 

This leads to a number of issues solved:

- The whole request mapping in `HttpRouterTester` is gone, and we don't need to worry about it duplicating code between the `RequestInitializer` and the test implementation.
- Validation errors are thrown in a place where the router can always catch it and send a response with validation error data — still need to do that. 
- We don't initialize the same request twice via the container, it's already registered when the router has found a matching route for the current URI
- And… yes, we _could_ support PSR-7 now without the user ever noticing. Maybe that makes me more open to the idea.